sdk/rust/src/consensus/zmq_driver.rs

```diff
@@ -74,13 +74,14 @@ pub fn start<T: AsRef<str>, E: Engine + Send + 'static >(endpoint: T, engine: E)
 
         let engine_thread = ::std::thread::spawn(move || {
             eprintln!("starting engine");
+            let (name, version) = { (engine.name(), engine.version()) };
             engine.start(
                 update_receiver,
                 Box::new(ZmqService::new(
                     validator_sender_clone,
                     ::std::time::Duration::from_secs(SERVICE_TIMEOUT),
-                    engine.name(),
```

Comment:
Is this a functional change?

Reply:
In order to borrow the engine as mutable here, it needs to not be borrowed anywhere else.